### PR TITLE
PR: automatic registration to iDVC

### PR DESCRIPTION
**Pull request by DanicaSTFC**
Closes issue #205 #173 #119 

- Modifies the method `LoadImagesAndCompleteRegistration` to expand the volumes by 2*registration box size in each direction. The size has been chosen to be 2 since 3 was slowing down the app and it is not necessary. The rounding works better with even numbers to locate point zero.
- Adds the method `enlarge_extent` to `MainWindow` to create the dimensions 2 times the registration box size.
- Adds the method `automatic_reg_run` to `MainWindow` to run the automatic registration class.
- Adds the file `class_automatic_registration` to the `utils` where the core method for automatic registration is implemented.
- Updates widgets in the registration tab with the values obtained from running perform_automatic_registration in the class automatic_registration.
- Adds a worker for the automatic registration where the signal result is passed to another method.
- Updates the viewer after the automatic reg has finished. 
- The signal finished is connected to a method.
- The buttons Reset and Set Automatic Registration are added.
- The manual registration can be implemented by the user after the automatic has finished.
- The help text is updated.
- The result from the worker and the viewer convention are consistent.
- Improve the conditions to exit the while loop in the class automatic registration.
- Change the behaviour when point zero is at the borders.
- Follow naming conventions.
- Add scipy to environment.
- Fixed full resolution cropping of the images.
- Add cancel button to reset and confirm registration.
- Fix automatic reg when the reg box size is changed in the same session for the full resolution.
- Run automatic yapf on automatic reg class.
- Checked build folder is uploading correctly with pip install.
- Add pop up dialog for finished registration, continue with the next tab.
- Delete some extra prints.

### CODE REVIEW

File: src/idvc/dvc_interface.py [hunk 1]
```diff
@@ -1825,7 +1838,48 @@ def translate_image_reg(self, *args, **kwargs):
...
+
+
+
+
+
+    def automatic_reg_run(self):
+
+        #get images full resolution and cropped 
+        data = [self.unsampled_ref_image_data,self.unsampled_corr_image_data]
+        [image0,image1]=[Converter.vtk2numpy(data[i]) for i in (0,1)]
```
Comment (paskino) [inline src/idvc/dvc_interface.py:1850]:
```suggestion
        [image0,image1]=[Converter.vtk2numpy(el) for el in data]
```

File: src/idvc/dvc_interface.py [hunk 2]
```diff
@@ -1825,7 +1838,48 @@ def translate_image_reg(self, *args, **kwargs):
...
+        print(image0.shape)
+        print(image1.shape)
+        target_x_extent = self.enlarge_extent(0)
+        target_y_extent = self.enlarge_extent(1)
+        print(target_x_extent,target_y_extent)
+        [image0,image1] = [image0[:,target_y_extent[0]:target_y_extent[1]+1,target_x_extent[0]:target_x_extent[1]+1],image1[:,target_y_extent[0]:target_y_extent[1]+1,target_x_extent[0]:target_x_extent[1]+1]]
```
Comment (paskino) [inline src/idvc/dvc_interface.py:1856]:
```suggestion
        [image0, image1] = [ el[:,target_y_extent[0]:target_y_extent[1]+1,target_x_extent[0]:target_x_extent[1]+1] for el in [image0, image1]]
```

**Comment by DanicaSTFC** [comment]
By testing the app in @paskino's machine, we found that 
- Scipy was missing from the environment.
- The automatic registration runs without errors both inside the image and at the borders.
- The results from the automatic registration are not correct when the images are uploaded at full resolution. This is a result of the way the coordinate system for point zero is evaluated and needs further investigation.
- The build folder is not updated correctly after pip install. This could have been fixed in master already and needs to be checked.

**Comment by DanicaSTFC** [comment]
@paskino and I are not completely sold on the "Reset" button but could not think of a better way to name it for now. the rest button is translating the registration to [0, 0, 0]

**Comment by paskino** [comment]
may close #173

**Comment by DanicaSTFC** [comment]
The help text has been changed. I opened an issue because I think this can be improved in general for the other tabs too but needs a bit of restructuring.

https://github.com/TomographicImaging/iDVC/issues/225

**Pull request merged**